iothub_client/inc/internal/iothubtransport_amqp_messenger.h

```diff
@@ -106,6 +106,9 @@ typedef struct AMQP_MESSENGER_CONFIG_TAG
 
     AMQP_MESSENGER_SUBSCRIPTION_CALLBACK on_subscription_changed_callback;
     void* on_subscription_changed_context;
+
+    pfTransport_GetOption_Product_Info_Callback prod_info_cb;
```

Comment:
>pfTransport_GetOption_Product_Info_Callback prod_info_cb; [](start = 4, length = 57)

Having 2 additional pointers in each header is unnecessary if we pass in the AMQP_DEVICE_HANDLE itself.

Reply:
Maybe in the future we will consolidate some things, but the device handle is not used everywhere.

---
In reply to: [295569174](https://github.com/Azure/azure-iot-sdk-c/pull/1056#discussion_r295569174) [](ancestors = 295569174)